### PR TITLE
PR - Journalist can create article

### PR DESCRIPTION
**Pull request by Ehugo2000**
[PT-story](https://www.pivotaltracker.com/story/show/176041267)
### User Story
```
As an API,
In order to keep a record of content,
I would like to be able to save a created article in a database.
```
## Changes proposed in this pull request:
* Adding Create action to articles_controller
* Creating category model
* Adding category in seeds.rb
## What I have learned working on this feature:
* Working with Postman
* Params
* Debugging
* Creating an article

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -10,4 +10,20 @@ def show
   rescue ActiveRecord::RecordNotFound => e
     render json: { message: 'Something went wrong, this article was not found' }, status: 404
   end
+
+  def create
+    article = Article.create(article_params)
+    if article.persisted?
+      render json: { message: 'Your article was successfully created' }, status: 201
+    else
+      render json: { message: 'Something went wrong!' }, status: 422
```
Comment (emtalen) [inline app/controllers/api/articles_controller.rb]: use the active record errors that you get if a title isn't added and so on. Put a debugger and get your way through to the specific errors.

File: spec/requests/api/journalist_can_create_an_article.spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+RSpec.describe 'POST/api/articles', type: :request do
+  let!(:articles) { create(:article) }
```
Comment (emtalen) [inline spec/requests/api/journalist_can_create_an_article.spec.rb]: why are you creating an article?

**Review by emtalen — CHANGES_REQUESTED** [review]

File: spec/requests/api/journalist_can_create_an_article.spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    it 'is expected to return a 422' do
+      expect(response).to have_http_status 422
+    end
+
+    it 'is expected to return a error message' do
+      expect(response_json['message']).to eq 'Something went wrong!'
+    end
+  end
+end
```
Comment (emtalen) [inline spec/requests/api/journalist_can_create_an_article.spec.rb]: you need to rename this file to _spec, that's why coveralls doesn't find it

File: spec/requests/api/journalist_can_create_an_article.spec.rb (hunk 2 above)
Comment (Ehugo2000) [reply spec/requests/api/journalist_can_create_an_article.spec.rb]: Dont we need it to create an Article in the test?

File: spec/requests/api/journalist_can_create_an_article.spec.rb (hunk 3 above)
Comment (Ehugo2000) [reply spec/requests/api/journalist_can_create_an_article.spec.rb]: Fixed

File: spec/requests/api/journalist_can_create_an_article.spec.rb (hunk 2 above)
Comment (Ehugo2000) [reply spec/requests/api/journalist_can_create_an_article.spec.rb]: it is removed now btw

Comment (emtalen) [reply spec/requests/api/journalist_can_create_an_article.spec.rb]: @Ehugo2000 only if you want to return an article, in this case you want to create it.

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
   namespace :api do
-    resources :articles, only: %i[index show]
+    resources :articles, only: %i[index show create]
```
Comment (ViktorHek) [inline config/routes.rb]: I think we have to change it to ```only: [:index, :show, :create]```. this works but the other way looks more readable.

File: spec/requests/api/journalist_can_create_an_article_spec.rb [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    before do
+      post '/api/articles',
+           params: {
+             article: {
+               title: 'A title',
+               lead: 'And a lead',
+               body: 'With a body',
+               category_id: category.id
+             },
+
```
Comment (KyunginNa) [inline spec/requests/api/journalist_can_create_an_article_spec.rb]: We can delete this additional line:)

File: config/routes.rb (hunk 4 above)
Comment (Ehugo2000) [reply config/routes.rb]: Fixed!

File: spec/requests/api/journalist_can_create_an_article_spec.rb (hunk 5 above)
Comment (Ehugo2000) [reply spec/requests/api/journalist_can_create_an_article_spec.rb]: Fixed!

File: spec/requests/api/journalist_can_create_an_article_spec.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  describe 'successfully creates an article' do
+    before do
+      post '/api/articles',
+           params: {
+             article: {
+               title: 'A title',
+               lead: 'And a lead',
+               body: 'With a body',
+               category_id: category.id
```
Comment (emtalen) [inline spec/requests/api/journalist_can_create_an_article_spec.rb:13]: How will you get the id of the category in the client? Would it be an easier solution for you in the client to sned in the label of the category and then here convert that by finding the category with that label and pass in that id when you save the article?
Just so you have thought this through and know how to deal with it in the client. This is a questions so I want an answer, there are no right or wrongs.

**Review by emtalen — CHANGES_REQUESTED** [review]

File: spec/requests/api/journalist_can_create_an_article_spec.rb (hunk 6 above)
Comment (ViktorHek) [reply spec/requests/api/journalist_can_create_an_article_spec.rb:13]: we are sending category_id from client

**Review by Ehugo2000 — COMMENTED** [review]
Looks good team!

**Review by emtalen — APPROVED** [review]

**Pull request merged**